de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/environment/GeneratedResourceFiles.java

```diff
@@ -0,0 +1,210 @@
+   *  If there is no internal folder (yet), the path will remain empty.
+   *  Since the absolute path to the directory may change, we only store the node directory name
+   *  and generate the path to the workflow on demand.
+   *  [relative path to the workflow] + [/node_dir] + "/internal" 
+   */
+  private String directoryName;
+
+
+  private String getBaseWorkflowDirectory() {
+    return NodeContext.getContext().getWorkflowManager().getWorkingDir().toString();
+  }
+
+  /**
+   * Main Constructor. Called by the Runner and FskObject nodes.
+   * The point is to store the name of the node, by which the model was executed,
+   * so all the files generated by the model can be found there.
+   * The name (+ id) of the node is unique for the workflow, so storing the name without
+   * the path is enough to reconstruct its absolute path later (e.g. in the Writer node). 
+   */
+  public GeneratedResourceFiles() {
+
+    // initialize list of resource file paths
+    this.resourceFiles = new ArrayList<>();
+
+    // get the name (+ id) of the currently executed node:
+    NodeContext nodeContext = NodeContext.getContext();
+
+    // Node ID build like this: [workflow]:[parentNode]:[actualNode]
+    // Node ID Example: 0:13:12
+    String nId = nodeContext.getNodeContainer().getID().toString();
+    String[] nodeIds= nId.split(":");
+
+    // get name of node and id to create the name of the node directory (e.g. "FSK Runner (#12)" )
+    String nodeName = nodeContext.getNodeContainer().getName() + " (#" + nodeIds[nodeIds.length - 1] + ")";
+
+    // since a node could be nested in a metanode/container: get the name of parent nodes
+    WorkflowManager wfm = nodeContext.getNodeContainer().getParent();
+    for(int i = nodeIds.length - 2; i > 0 ; i--) {
+
+      nodeName = wfm.getName()+" (#" + nodeIds[i] + ")" + File.separator + nodeName ;
+      wfm = wfm.getParent();
+    }
+
+
+    this.directoryName = nodeName;
+
+
+
+  }
+
+
+
+  /**
+   * Adds a file(absolute path) generated by the model execution (FSK Runner) to the list. 
+   * */
+  public void addResourceFile(File file){
+
+    String resourceFolder = "GeneratedResources";
+
+    try {
+
+      File temp_dir = FileUtil.createTempDir(resourceFolder);
+      final File resourceFile = new File(temp_dir.getAbsolutePath(),file.getName());
+
+      FileUtil.copy(file, resourceFile );
+
+      this.resourceFiles.add(resourceFile.getAbsolutePath());
+
+    }catch(Exception e) {
+      System.out.println("resource File Error: " + e);
+    }
+
+  }
+
+  /**
+   * Returns a list of file paths pointing to the files that were generated by the model previously.
+   * The files might be in a temporary folder or in the /internal folder of the Runner node that created the files.
+   *     
+   * */ 
+  public List<Path> getResourcePaths(){
+
+    List<Path> resourcePaths = new ArrayList<>();
+
+    this.resourceFiles.forEach(f -> resourcePaths.add(Paths.get(f)));

```

Comment:
This can be written with an stream:
```java
List<Path> resourcePaths = this.resourceFiles.stream().map(Paths::get).collect(Collectors.toList());
```